### PR TITLE
Upgrade changes the column name

### LINKED ISSUE #1731: Error column name after upgrade
### Steps to reproduce the behavior (Required)
1. create table in starrocks version 1.17.4
```
CREATE TABLE `dm_hfc_research_follow_status` (
  `id` int(11) NULL COMMENT "",
  `pid` int(11) NULL COMMENT "",
  `orgid` int(11) NULL COMMENT "",
  `org` varchar(200) NULL COMMENT "",
  `cases_status` tinyint(4) NULL COMMENT "",
  `HF0312` varchar(1) NULL COMMENT "",
  `HF031201` varchar(50) NULL COMMENT ""
) ENGINE=OLAP 
UNIQUE KEY(`id`, `pid`)
COMMENT ""
DISTRIBUTED BY HASH(`pid`) BUCKETS 18 
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);
```
2. insert data '...'
3. do schema change
```
alter table dm_hfc_research_follow_status modify column HF031201 varchar(64);
```
4. upgrade starrocks to version 1.19 after schema change is finished;
5. desc dm_hfc_research_follow_status;
### Expected behavior (Required)
+--------------+--------------+------+-------+---------+---------+
| Field        | Type         | Null | Key   | Default | Extra   |
+--------------+--------------+------+-------+---------+---------+
| id           | INT          | Yes  | true  | NULL    |         |
| pid          | INT          | Yes  | true  | NULL    |         |
| orgid        | INT          | Yes  | false | NULL    | REPLACE |
| org          | VARCHAR(200) | Yes  | false | NULL    | REPLACE |
| cases_status | TINYINT      | Yes  | false | NULL    | REPLACE |
| HF0312       | VARCHAR(1)   | Yes  | false | NULL    | REPLACE |
| HF031201     | VARCHAR(64)  | Yes  | false | NULL    | REPLACE |
+--------------+--------------+------+-------+---------+---------+
### Real behavior (Required)
+--------------+--------------+------+-------+---------+---------+
| Field        | Type         | Null | Key   | Default | Extra   |
+--------------+--------------+------+-------+---------+---------+
| id           | INT          | Yes  | true  | NULL    |         |
| pid          | INT          | Yes  | true  | NULL    |         |
| orgid        | INT          | Yes  | false | NULL    | REPLACE |
| org          | VARCHAR(200) | Yes  | false | NULL    | REPLACE |
| cases_status | TINYINT      | Yes  | false | NULL    | REPLACE |
| HF0312       | VARCHAR(1)   | Yes  | false | NULL    | REPLACE |
| __doris_shadow_HF031201     | VARCHAR(64)  | Yes  | false | NULL    | REPLACE |
+--------------+--------------+------+-------+---------+---------+

### StarRocks version (Required)
 - upgrade starrocks from version 1.17.4 to version 1.19

### PR DESCRIPTION
**Pull request by sevev**
Before version 1.18, starrocks use `__doris_shadow_` prefix to do schema change.
After version 1.18, starrocks use `__starrocks_shadow_` prefix to do schema change.
Starrocks after version 1.18 can't identify `__doris_shadow_` prefix and will handle it as normal column name.
Add `__doris_shadow_` prefix check to fix it

will close https://github.com/StarRocks/starrocks/issues/1731

### CODE REVIEW

**Comment by sevev** [comment]
run starrocks_fe_unittest

**Comment by imay** [comment]
@sevev 
Please write the reason of this bug. And how you fix it in the comment message.

**Review by imay — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -109,6 +109,7 @@
 
     // all shadow indexes should have this prefix in name
     public static final String SHADOW_NAME_PRFIX = "__starrocks_shadow_";
+    public static final String SHADOW_NAME_PRFIX_V1 = "__doris_shadow_";
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:112]: Better to leave some comment for the reason why it is here.

**Review by imay — APPROVED** [review]
LGTM

**Review by chaoyli — APPROVED** [review]

**Pull request merged**